sdks/java/io/rabbitmq/src/main/java/org/apache/beam/sdk/io/rabbitmq/RabbitMqIO.java

```diff
@@ -475,6 +489,68 @@ public boolean advance() throws IOException {
       return true;
     }
 
+    /**
+     * ake delivery serializable by cloning all non-serializable values into serializable ones. If
+     * it is not possible, initial delivery is returned and error message is logged
+     *
+     * @param processed
+     * @return
+     */
+    private Delivery serializableDeliveryOf(Delivery processed) {
```

Comment:
I would rather move this logic into the `RabbitMqMessage`, as it's where we are dealing with serialization.

Reply:
it's indeed better